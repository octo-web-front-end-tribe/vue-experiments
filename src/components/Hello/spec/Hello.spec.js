import {shallow} from 'vue-test-utils';
import Hello from '../Hello';

describe('Hello.vue', () => {
  it('should match snapshot', () => {
    // when
    const wrapper = shallow(Hello);

    // then
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('h1', () => {
    it('should contain default message', () => {
      // when
      const wrapper = shallow(Hello);

      // then
      const title = wrapper.find('h1');
      expect(title.text()).toContain('Welcome'); // default
    });

    it('should update when msg data is changed', () => {
      // when
      const wrapper = shallow(Hello);

      // then
      const title = wrapper.find('h1');
      wrapper.setData({msg: 'kikoo'});
      expect(title.text()).toContain('kikoo');
    });

    it('should contain name in props', () => {
      // given
      const props = {name: 'fake name'};

      // when
      const wrapper = shallow(Hello, {propsData: props});

      // then
      const title = wrapper.find('h1');
      expect(title.text()).toContain('fake name');
    });
  });
});
