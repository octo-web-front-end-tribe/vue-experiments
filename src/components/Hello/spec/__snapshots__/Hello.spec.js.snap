// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello.vue should match snapshot 1`] = `
<div
  class="hello"
>
  <h1>
    Welcome 
  </h1>
</div>
`;
